### ISSUE TITLE
[0.9]Server Timeout Duct Caching

### ISSUE THREAD

**Issue by SonOfTheStars** [RS485/LogisticsPipes#668]
References to https://github.com/RS485/LogisticsPipes/issues/660
If I add more than 25 connections via Ducts only, the Server starts Timing me out and skipping a bunch of Ticks.
This happens each time you add anything to the network.
Ill profile this later today.

Runtime for caching seems to grow exponentially again.

**Comment by Bitterholz** [comment]
Runtime for caching seems to grow exponentially again.

**Comment by davboecki** [comment]
There isn't realy anything that can be done except limit the amount of routed pipes that can be connected trough an unrouted network. I tried to spread the load over several ticks but that didn't realy help and caused more trouble. [The attempt can be found here: https://github.com/RS485/LogisticsPipes/commit/0d3c9a06cf7dc9e6f85c951e96422b228c543e85]

**Comment by davboecki** [comment]
https://github.com/RS485/LogisticsPipes/commit/ea822ae81e67df0f467d6a564f6df378836d01ca Will NOT fix the performance. It just makes using more than 16 pipes on one duct system useless, becasue any amount over that would cause the performance problems.

**Comment by SonOfTheStars** [comment]
Thanks for trying @davboeki!
Ill now try to do warp Pipes as LP internal ones

**Comment by davboecki** [comment]
@Bitterholz Can you test this build: https://dl.dropboxusercontent.com/u/33855125/logisticspipes-0.9.0.TDPerformanceTest.jar . It contains several permance improvemnts and should allow you to connect more Item ducts to LP routed pipes. The Limit is set to 64 routed pipes on an ItemDuct network inside that build.

**Comment by SonOfTheStars** [comment]
What in particular did you change @davboecki? Ill test it Right now

**Comment by SonOfTheStars** [comment]
Allright, i could not notice any performance Hits except a tiny itty bitty render lag when I added loads of Pipes at once from a prebuild system.

Whats up with the Red Sparkles the Pipes emit every now and then or if you rightclick them?

**Comment by davboecki** [comment]
The light red and light green particles now indicate adjacent checks and routing table updates.

I changed several things to improve the performance. I fixed a couple of double calls to the check adj code. I moved the LSA update into the routing thread to get rid of the writeLock inside the ADJ check. That one caused some performance issues. I also tought the PathFinder how to use TD's internal Routing information to determine the routes faster. 

All inside that commit: https://github.com/RS485/LogisticsPipes/commit/56aff51c0796b751185e123d051a5b456148db6d
